lang/tests/src/test/scala/com/wavesplatform/lang/IntegrationTest.scala

```diff
@@ -2174,4 +2174,88 @@ class IntegrationTest extends PropSpec with PropertyChecks with ScriptGen with M
     genericEval[Environment, EVALUATED](script2, ctxt = v5Ctx, version = V5, env = utils.environment) should
       produce("Alias name length=31 exceeds limit=30")
   }
+
+  property("unicode broken") {
+    val ver = V4
+
+    val script1 = s"""take("x冬x", 2)"""
+    eval(script1, version = ver) shouldBe
+      Right(CONST_STRING("x\ud87e").explicitGet())
+
+    val script2 = s"""size("x冬x")"""
+    eval(script2, version = ver) shouldBe
+      Right(CONST_LONG(4))
+
+    val script3 = s"""drop("x冬x", 2)"""
+    eval(script3, version = ver) shouldBe
+      Right(CONST_STRING("\udc1ax").explicitGet())
+
+    val script4 = s"""takeRight("x冬x", 2)"""
+    eval(script4, version = ver) shouldBe
+      Right(CONST_STRING("\udc1ax").explicitGet())
+
+    val script5 = s"""dropRight("x冬x", 2)"""
+    eval(script5, version = ver) shouldBe
+      Right(CONST_STRING("x\ud87e").explicitGet())
+  }
+
+  property("unicode indexOf") {
+    val src =
+      """ "x冬xqweqwe".indexOf("we") """
+    genericEval[Environment, EVALUATED](src, ctxt = v5Ctx, version = V5, env = utils.environment) shouldBe Right(CONST_LONG(4L))
```

Comment:
What this function returns in V4 context? 
Also the funny script that may be tested
```
indexOf(take("世界x冬x", 4), take("冬", 1))
```
What it returns for V4 and V5?

Reply:
All characters are 2-bytes. It returns 3 in both cases.